helpers/kernel_symbols.go

```diff
@@ -0,0 +1,102 @@
+package helpers
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+/*
+ * This package gives the ability to hold all the known kernel symbols.
+ * the package parse the /proc/kallsyms file that hold the known kernel symbol
+ *
+ * The KernelSymbolTable type holds map of all the kernel symbols with a key which is the kernel object owner and the name with under-case between them
+ * which means that symbolMap looks like [objectOwner_objectname{SymbolData}, objectOwner_objectname{SymbolData}, etc...]
+ * the key naming is because sometimes kernel symbols can have the same name or the same address which prevents to key the map with only one of them
+ *
+ */
+
+type KernelSymbolTable struct {
+	symbolMap   map[string]KernelSymbol
+	initialized bool
+}
+
+type KernelSymbol struct {
+	Name    string
+	Type    string
+	Address uint64
+	Owner   string
+}
+
+func (k *KernelSymbolTable) IsInTextSegment(addr uint64) (bool, error) {
```

Comment:
Since the symbol table itself is the method receiver I think it makes more sense for the method to be named `ContainsAddr`

Reply:
I think it better to create another function that will be member of `KernelSymbol struct` that will can compare the address, like that one:
`func (k *KernelSymbol) ContainsAddr(boundreyLowerAddr uint64, boundreyUpperAddr uint64) (bool, error) `

and ill change `IsInTextSegment` to use the ContainsAddr, like that.

what you think? @grantseltzer

Reply:
I don't agree. I feel that the method name should be grammatically based off the method receiver. For example:

`KernelSymbolTable.IsInTextSegment()` doesn't make sense to me because you're not checking if the kernel symbol table is in the text segment, you're checking if the passed address is in the text segment. 

By the same logic, having `KernelSymbol.ContainsAddr()` doesn't make sense because you're not checking if the kernel symbol contains the address, you're checking if the kernel symbol is within the range of passed addresses. 

That's why I suggested renaming it to `ContainsAddr`, I believe you're checking if the `KernelSymbolTable` contains the address.

Reply:
I'll add that it's fine to have another helper function, but I think the name should be `KernelSymbol.IsInAddressRange` or something like that.

Reply:
Definitely Agree  (:

Your explain of the `KernelSymbolTable.IsInTextSegment()` and `KernelSymbol.ContainsAddr()` made me understand your idea, thanks! 

One little Q:

If ill rename it to be `KernelSymbolTable.ContainsAddr()', I will need to change the logic to only check for the address in the `kerenlSymbolTable` right?

So in that case ill have to create another function that check for address range , like you wrote about the `KernelSymbol.IsInAddressRange`

also I have the `GetSymbolByAddr` that check for symbol byt its address, isn't the same as the `containsAddr` idea?